src/granite/error.cr

```diff
@@ -1,9 +1,15 @@
+require "json"
+
 class Granite::Error
   property field, message
 
   def initialize(@field : (String | Symbol | JSON::Any), @message : String? = "")
   end
 
+  def to_json
```

Comment:
You actually want to define a `def to_json(builder : JSON::Builder)` then use the `builder` object to create the object.

See https://crystal-lang.org/api/master/JSON/Builder.html

Reply:
@Blacksmoke16 For my own learning, what are the advantages of using the JSON Builder over having just a `to_json` method with no params?

Reply:
Essentially the same idea as why you define `to_s(io : IO)` instead of `to_s`.

See https://github.com/crystal-lang/crystal/blob/master/src/json/to_json.cr#L2-L12